src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/JakartaCodeActionHandler.java

```diff
@@ -57,8 +57,8 @@ public List<CodeAction> codeAction(JakartaJavaCodeActionParams params, IPsiUtils
             int start = DiagnosticsHelper.getStartOffset(unit, params.getRange(), utils);
             int end = DiagnosticsHelper.getEndOffset(unit, params.getRange(), utils);
             var mpParams = new MicroProfileJavaCodeActionParams(params.getTextDocument(), params.getRange(), params.getContext());
-            JavaCodeActionContext context = new JavaCodeActionContext(unit, start, end - start, utils,
-                    mpParams);
+            JavaCodeActionContext context = new JavaCodeActionContext(unit.getViewProvider().clone().getPsi(unit.getLanguage()),
```

Comment:
Might be wise to add a comment here on the importance of cloning the PsiFile.

Reply:
it seems JavaCodeActionContext already has a `.copy()` method that we can use, which uses this same method of cloning the `PsiFile.getViewProvider()`

Reply:
Good idea.